include/class.thread.php

```diff
@@ -993,14 +993,19 @@ function lookupByEmailHeaders(&$mailinfo, &$seen=false) {
                 //Lookup by ticket number
                 && ($ticket = Ticket::lookupByNumber($match[1]))
                 //Lookup the user using the email address
-                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))) {
+                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))
+                && ($staff = Staff::getIdByEmail($mailinfo['email']))) {
```

Comment:
Shouldn't this be something like
```php
if ( ... 
    && (
           ($user = User::lookup(array('emails__address' => $mailinfo['email'])))
        || ($staff = Staff::getIdByEmail($mailinfo['email']))
    )
) {
```
Since it's unlikely that the user email will have a hit for both a User and a Staff?